### DISCUSSION TITLE
Allow accessing OAuth tokens

### DISCUSSION THREAD

**Discussion by sabinehoi** [directus/directus#4563] [Archive]
After the user signs-in with OAuth, how can I access the service refresh token or access token provided by the provider's `/token` endpoint?

I am trying to do this so I can use APIs on behalf of the user (eg. [Miscrosoft Graph OAuth](https://docs.microsoft.com/en-us/graph/auth-v2-user#4-use-the-access-token-to-call-microsoft-graph)).

Maybe "inject" it into the "req" object or similar for use in custom endpoints? This can also be done if a hook was provided for the oauth callback that recieves the token.

**Comment by rijkvanzanten** [comment]
Heya! The oAuth token for the third party isn't persisted on the user or in the users session token, so there's no way to use it later (in a custom hook/endpoint f.e.) at this point

**Reply by sabinehoi** [reply]
Hey, thanks for the quick reply! Do you see this as an important feature, and if so, any ideas where to start (happy to submit a PR)? I was thinking I could implement oauth hooks? Devs then keep track of the tokens by intercepting it in a hook.

**Reply by rijkvanzanten** [reply]
I don't think having "native" support for tracking this oauth token for the duration of the session has any importance as of right now, but your suggestion to add a "on oauth login" hook that you can then use to do whatever makes total sense to me 👍🏻

**Reply by Nisthar** [reply]
> I don't think having "native" support for tracking this oauth token for the duration of the session has any importance as of right now, but your suggestion to add a "on oauth login" hook that you can then use to do whatever makes total sense to me 👍🏻

The oauth token is needed to get more info from the provider like avatar Right?

**Reply by rijkvanzanten** [reply]
@Nisthar this discussion is over 2 years old. The whole auth system and extensions engine have been replaced since, so no clue if anything from above is still relevant. I'd recommend opening a new discussion if you have any specific questions around the current version :)
